test/params_test.c

```diff
@@ -655,14 +657,56 @@ static int check_int_from_text(const struct int_from_text_test_st a)
     return a.expected_res;
 }
 
+static int check_octetstr_from_hexstr(void)
+{
+    OSSL_PARAM param;
+    static const char *values[] = { "", "F", "FF", "FFF", "FFFF", NULL };
+    int i;
+    int errcnt = 0;
+
+    /* Test odd vs even number of hex digits */
+    for (i = 0; values[i] != NULL; i++) {
+        int expected = (strlen(values[i]) % 2) != 1;
+        int result;
+
+        ERR_clear_error();
+        memset(&param, 0, sizeof(param));
+        if (expected)
+            result =
+                TEST_true(OSSL_PARAM_allocate_from_text(&param,
```

Comment:
I'd normally use `result` and `expected` for the value from the function being tested.

This all works, but has two calls to the function under test, which meant checking that the parameters were the same, and it took me longer than I expected to read this correctly.

I would probably have written something along the lines of (untested; not showing clear_error or memset; and possibly better with `TEST_int_eq()`)

```
        int expected = (strlen(values[i]) % 2) == 0;
        result = OSSL_PARAM_allocate_from_text(&param,
                                               params_from_text,
                                               "hexoctets", values[i], 0,
                                               NULL);
        if (!TEST_true(result == expected)) {
            TEST_error("unexpected OSSL_PARAM_allocate_from_text() %s for 'octets' \"%s\"",
                       (expected ? "success" : "failure"), values[i]);
            errcnt++;
        }
```

Reply:
Point is that the `TEST_` macros work a bit like assert(), and displays the expression that was tested if it doesn't return what the test macro expected.  It makes more immediate sense to me if it then displays that `OSSL_PARAM_allocate_from_text(...)` didn't return true, than if it displays that `result == expected` isn't true.

That is also a reason why I have an explicit `TEST_false()` alternative, because false is exactly what I expect in that case, and if `OSSL_PARAM_allocate_from_text()` returns true in that case, then that should be displayed as such.